src/dataflow-types/src/client/controller/compute.rs

```diff
@@ -353,13 +354,37 @@ where
                 source_imports.insert(id, desc);
             }
 
+            let mut sink_exports = BTreeMap::new();
+            for (id, se) in d.sink_exports {
```

Comment:
I quickly tried that, but it gets unwieldy because the early return here:
```
let metadata = self.storage_controller.collection_metadata(id)?;
```

... isn't possible anymore when we are inside a `map`. It can be made to work of course, but I think the loop-based version would end up more readable.

Reply:
Thanks, this makes sense!